services/background-worker/src/Wire/Defederation.hs

```diff
@@ -53,75 +50,65 @@ deleteFederationDomainInner' go (msg, envelope) = do
 -- What should we do with non-recoverable (unparsable) errors/messages?
 -- should we deadletter, or do something else?
 -- Deadlettering has a privacy implication -- FUTUREWORK.
-deleteFederationDomainInner :: (RabbitMQEnvelope e) => (Q.Message, e) -> AppT IO ()
-deleteFederationDomainInner (msg, envelope) = do
-  env <- ask
-  let manager = httpManager env
-      req :: Domain -> Request
-      req dom =
-        defaultRequest
-          { method = methodDelete,
-            secure = False,
-            host = galley env ^. epHost . to encodeUtf8,
-            port = galley env ^. epPort . to fromIntegral,
-            path = "/i/federation/" <> toByteString' dom,
-            requestHeaders = ("Accept", "application/json") : requestHeaders defaultRequest,
-            responseTimeout = defederationTimeout env
-          }
-  let callGalley d = do
-        -- Retry the request a couple of times. If the final one fails, catch the exception\
-        -- so that we can NACK the message and requeue it.
-        resp <- try $ recovering x3 httpHandlers $ \_ -> liftIO $ httpLbs (req d) manager
-        either
-          -- Requeue the exception and rethrow the exception
-          (\(e :: SomeException) -> liftIO (reject envelope True) >> throwM e)
-          go
-          resp
+deleteFederationDomainInner :: (RabbitMQEnvelope e) => MVar () -> (Q.Message, e) -> AppT IO ()
+deleteFederationDomainInner runningFlag (msg, envelope) =
   deleteFederationDomainInner' (const callGalley) (msg, envelope)
   where
-    go :: Response L.ByteString -> AppT IO ()
-    go resp = do
-      let code = statusCode $ responseStatus resp
-      if code >= 200 && code <= 299
-        then do
-          liftIO $ ack envelope
-        else -- ensure that the message is requeued
-        -- This message was able to be parsed but something
-        -- else in our stack failed and we should try again.
-          liftIO $ reject envelope True
+    callGalley d = do
+      env <- ask
+      -- Jittered exponential backoff with 10ms as starting delay and 300s as max
+      -- delay. When 300s is reached, every retry will happen after 300s.
+      let policy = capDelay 300_000_000 $ fullJitterBackoff 10000
```

Comment:
```suggestion
      -- Jittered exponential backoff with 10ms as starting delay and 300s as max
      -- delay. When 60s is reached, every retry will happen after 300s.
      let policy = capDelay 60_000_000 $ fullJitterBackoff 10000
```

I think 5 mins is too much to wait for a local component

Reply:
Done